cleanup-anaconda.sh

```diff
@@ -3,35 +3,90 @@
 source $GITHUB_WORKSPACE/.github/scripts/common.sh
 # `for ... in $(anaconda ...` fails silently if there's any problem with anaconda
 source $GITHUB_WORKSPACE/.github/scripts/test_anaconda.sh
+
+set -e
 set -x
 
-#if the timestamp is older than one week, remove the whole label
+#print cleanup summary before exiting
+log_file=$GITHUB_WORKSPACE/cleanup.log
+function print_log {
+    end_section  # in case exiting inside a section
+    start_section "Cleanup summary"
+    cat $log_file
+    end_section
+}
+trap print_log EXIT
+
+#packages will be removed if their label is older than 'ago'
 ago="7 days ago"
+#packages won't be removed if they were downloaded more than 'download_threshold' times
+download_threshold=30
```

Comment:
This might cause some packages to never be cleaned.
For example, `isl` for Linux seems to have 24 downloads right after the CI is finished: https://anaconda.org/LiteX-Hub/isl/files

This is an extreme example but there are at least a few packages with 3+ downloads so I'd go for at least 5-10.
What do you think?